.pipelines/singletenancy/cilium-nodesubnet/cilium-nodesubnet-e2e-step-template.yaml

```diff
@@ -0,0 +1,196 @@
+parameters:
+  name: ""
+  clusterName: ""
+  scaleup: ""
+
+steps:
+  - task: AzureCLI@2
+    displayName: 'Update IP configs'
+    inputs:
+      azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+      scriptLocation: "inlineScript"
+      scriptType: "bash"
+      addSpnToEnvironment: true
+      inlineScript: |
+          set -e
+          clusterName=${{ parameters.clusterName }}
+          secondary_config_count=64
+          RESOURCE_GROUP="MC_${clusterName}_${clusterName}_$(REGION_AKS_CLUSTER_TEST)"
+          VMSS_NAME=$(az vmss list -g "$RESOURCE_GROUP" --query '[0].name' -o tsv)
+          if [ -z "$VMSS_NAME" ]; then
+              echo "Failed to get VMSS name."
+              exit 1
+          fi
+
+          vmss_info=$(az vmss show -g "$RESOURCE_GROUP" -n "$VMSS_NAME")
+
+          secondary_configs=()
+
+          network_profile=$(echo "$vmss_info" | jq -r '.virtualMachineProfile.networkProfile')
+
+          if [ "$network_profile" != "null" ]; then
+              network_interface_configurations=$(echo "$network_profile" | jq -c '.networkInterfaceConfigurations[]')
+              for nic_config in $network_interface_configurations; do
+                  primary_ip_config=$(echo "$nic_config" | jq -c '.ipConfigurations[] | select(.primary == true)')
+
+                  if [ ! -z "$primary_ip_config" ]; then
+                      for ((i=2; i<=secondary_config_count+1; i++)); do
+                          ip_config_name="ipconfig$i"
+                          new_ip_config=$(echo "$primary_ip_config" | jq --arg name "$ip_config_name" '.name = $name | .primary = false')
+                          secondary_configs+=("$new_ip_config")
+                      done
+                  fi
+
+                  updated_ip_configs=$(echo "$nic_config" | jq --argjson new_configs "$(printf '%s\n' "${secondary_configs[@]}" | jq -s '.')" '.ipConfigurations += $new_configs')
+                  network_profile=$(echo "$network_profile" | jq --argjson updated_nic "$updated_ip_configs" '.networkInterfaceConfigurations |= map(if .name == $updated_nic.name then $updated_nic else . end)')
+              done
+          fi
+
+          az vmss update -g "$RESOURCE_GROUP" -n "$VMSS_NAME" --set virtualMachineProfile.networkProfile="$network_profile"
+          az vmss update-instances -g "$RESOURCE_GROUP" -n "$VMSS_NAME" --instance-ids '*'
+
+  - bash: |
```

Comment:
Want to confirm if there is anything different we are testing in the yaml for cilium ? Seems like we can re-use the existing code of pipeline. Otherwise we have to maintain/update scenarios at multiple places(if not already doing it)

Reply:
Cilium installation can be reused, but there are indeed some differences between existing E2E scenarios and this one (see INSTALL_CNS_NODESUBNET). The subsequent stages will probably stay the same, though I am not a 100% sure just yet.

Reply:
I just moved the common tests from Azilium and this PR into a template file. We may change that as this work progresses and we find new things to test. For now we can avoid the redundant code.